src/go/rpk/pkg/cli/container/start.go

```diff
@@ -477,3 +493,44 @@ func nodeAddr(port uint) string {
 		port,
 	)
 }
+
+// getContainerErr attempts to fetch the latest stderr output from the first
+// Redpanda node. It may reveal reasons for failing to start.
+func getContainerErr(state *common.NodeState, c common.Client) (string, error) {
+	ctx, _ := common.DefaultCtx()
+
+	json, err := c.ContainerInspect(ctx, state.ContainerID)
+	if err != nil {
+		return "", fmt.Errorf("could not inspect container: %v", err)
+	}
+
+	reader, err := c.ContainerLogs(
+		ctx,
+		state.ContainerID,
+		container.LogsOptions{
+			ShowStdout: false,
+			ShowStderr: true,
+			Since:      json.State.StartedAt,
+		},
+	)
+	if err != nil {
+		return "", fmt.Errorf("could not get container logs: %v", err)
+	}
+
+	// Docker logs over the wire are multiplexed using stdcopy package. To
+	// demux this stream we need to use stdcopy.StdCopy. See:
+	// https://github.com/moby/moby/issues/32794#issuecomment-297151440
+	bErr := bytes.NewBufferString("")
+	_, err = stdcopy.StdCopy(bErr, bErr, reader)
+	if err != nil {
+		return "", fmt.Errorf("unable to read docker logs: %v", err)
+	}
+
+	scanner := bufio.NewScanner(bErr)
+	var errStr string
+	for scanner.Scan() {
+		errStr += scanner.Text() + "\n"
+	}
+
+	return errStr, nil
```

Comment:
Why is this needed? This is reading the buffer line by line, and allocating each line, and then reallocating to add a newline, to append to a string to return what looks to be the same text as the input buffer -- isn't this equivalent to `return bErr.String(), nil`?

Reply:
Not needed, it was from the previous code and I didn't realize that I could take it out after copying it. Thanks!